UnityProject/Assets/Ardity/Scripts/Threads/AbstractSerialThread.cs

```diff
@@ -255,7 +260,17 @@ private void RunOnce()
                 }
                 else
                 {
-                    Debug.LogWarning("Queue is full. Dropping message: " + inputMessage);
+                    object droppedMessage;
+                    if (dropOldMessage)
+                    {
+                        droppedMessage = inputQueue.Dequeue();
+                        inputQueue.Enqueue(inputMessage);
+                    }
+                    else
+                    {
+                        droppedMessage = inputMessage;
+                    }
+                    Debug.LogWarning("Queue is full. Dropping message: " + droppedMessage);
```

Comment:
In general, I think the idea of this change is good, but this should be the default instead of being a configurable option.